File: python/summarize_day.py
import os
from pathlib import Path
import sys
import os.path

from lib.client import get_lifelogs
from lib.llm import summarize_lifelogs
from lib.env import load_env

# Add the parent directory to the path so we can import constants
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

# Import .env files
root_dir = Path(__file__).resolve().parent.parent
load_env(root_dir)

def main():
    # Get transcripts, limiting size because OpenAI has a 128k context window
    lifelogs = get_lifelogs(limit=10)

    # Summarize transcripts
    summarize_lifelogs(lifelogs)

if __name__ == "__main__":
    main() 

File: typescript/summarize_day.ts
import dotenv from "dotenv";
import OpenAI from "openai";
import { getLifelogs } from "./_client";

// Load environment variables
dotenv.config();

async function summarizeLifelogs(
  lifelogs: any[],
  shouldStream = true
): Promise<void | string> {
  const client = new OpenAI({
    apiKey: process.env.OPENAI_API_KEY,
  });

  if (shouldStream) {
    const stream = await client.chat.completions.create({
      model: "gpt-4.1",
      messages: [
        {
          role: "system",
          content: "You are a helpful assistant that summarizes transcripts.",
        },
        {
          role: "user",
          content: `Summarize the following transcripts: ${JSON.stringify(
            lifelogs
          )}`,
        },
      ],
      stream: true,
    });

    for await (const chunk of stream) {
      if (chunk.choices[0]?.finish_reason === null) {
        process.stdout.write(chunk.choices[0]?.delta?.content || "");
      }
    }
  } else {
    const response = await client.chat.completions.create({
      model: "gpt-4.1",
      messages: [
        {
          role: "system",
          content: "You are a helpful assistant that summarizes transcripts.",
        },
        {
          role: "user",
          content: `Summarize the following transcripts: ${JSON.stringify(
            lifelogs
          )}`,
        },
      ],
      stream: false,
    });

    const content = response.choices[0]?.message?.content;
    return content || "";
  }
}

async function main(): Promise<void> {
  try {
    // Get transcripts, limiting size because OpenAI has a 128k context window
    const lifelogs = await getLifelogs({
      apiKey: process.env.LIMITLESS_API_KEY || "",
      limit: 10,
    });

    // Summarize transcripts
    await summarizeLifelogs(lifelogs);
  } catch (error) {
    console.error("Error:", error);
    process.exit(1);
  }
}

// Run the main function
main();
